python/cudf/cudf/core/reshape.py

```diff
@@ -231,27 +232,71 @@ def concat(objs, axis=0, join="outer", ignore_index=False, sort=None):
       letter  number  animal    name
     0      a       1    bird   polly
     1      b       2  monkey  george
+
+    Combine a dictionary of DataFrame objects horizontally:
+
+    >>> d = {'first': df1, 'second': df2}
+    >>> cudf.concat(d, axis=1)
+      first           second
+      letter  number  letter  number
+    0      a       1       c       3
+    1      b       2       d       4
     """
     # TODO: Do we really need to have different error messages for an empty
     # list and a list of None?
     if not objs:
         raise ValueError("No objects to concatenate")
 
-    objs = [obj for obj in objs if obj is not None]
-
-    if not objs:
-        raise ValueError("All objects passed were None")
-
     axis = _AXIS_MAP.get(axis, None)
     if axis is None:
         raise ValueError(
             f'`axis` must be 0 / "index" or 1 / "columns", got: {axis}'
         )
 
+    if isinstance(objs, dict):
+        if axis != 1:
```

Comment:
The docs say we need to support `"columns"` as well as `1`. Can we add that, with tests?

Reply:
That's handled in line 250 above, I think?

Reply:
wence- is correct, though i added it to the tests found here nonetheless: https://github.com/rapidsai/cudf/pull/15623/commits/516d28bfd498206bafd9ee45cf593d5ad2810ab7

feel free to remove it from the tests if you feel it's unnecessasry

Reply:
I didn’t scroll through enough context. :) Thanks for the tests!